### PR TITLE
Changes needed for tests to pass on my desktop

### PR DESCRIPTION
**Pull request by jkausrelay**
- Two changes needed to make all the of the tests pass on my desktop
- tests/test_free_energy.py::test_functional is related to single vs double precision
-  tests/test_endpoint_correction.py::test_endpoint_correction seems to be a convergence issue at k=1000. I can either increase n_steps*=3 (2 minutes slower) or increase the overlap cutoff value to get it to pass.

### CODE REVIEW

File: tests/test_free_energy.py [hunk 1]
```diff
@@ -317,14 +317,15 @@ def test_functional():
     box = np.eye(3) * 100
     lam = 0.5
 
-    for precision in [np.float32, np.float64]:
+    tol_at_precision = {np.float32: 2.5e-10, np.float64: 1e-10}
+    for precision, tol in tol_at_precision.items():
```
Comment (maxentile) [inline tests/test_free_energy.py:321]: Surprised this caused an issue locally but not on the test runner... Regardless, it's a good idea to have separate tolerances for the different precision modes!

**Review by maxentile — APPROVED** [review]
Good catches!

**Comment by mcwitt** [comment]
@jkausrelay would you mind posting the output of `nvidia-smi` on your desktop for reference?

**Comment by jkausrelay** [comment]
> @jkausrelay would you mind posting the output of `nvidia-smi` on your desktop for reference?

(base) kaus@desktop:~$ nvidia-smi
Tue Mar 22 08:29:55 2022       
+-----------------------------------------------------------------------------+
| NVIDIA-SMI 510.47.03    Driver Version: 510.47.03    CUDA Version: 11.6     |
|-------------------------------+----------------------+----------------------+
| GPU  Name        Persistence-M| Bus-Id        Disp.A | Volatile Uncorr. ECC |
| Fan  Temp  Perf  Pwr:Usage/Cap|         Memory-Usage | GPU-Util  Compute M. |
|                               |                      |               MIG M. |
|===============================+======================+======================|
|   0  NVIDIA RTX A4000    Off  | 00000000:03:00.0  On |                  Off |
| 41%   29C    P8     8W / 140W |    109MiB / 16376MiB |      0%      Default |
|                               |                      |                  N/A |
+-------------------------------+----------------------+----------------------+
                                                                               
+-----------------------------------------------------------------------------+
| Processes:                                                                  |
|  GPU   GI   CI        PID   Type   Process name                  GPU Memory |
|        ID   ID                                                   Usage      |
|=============================================================================|
|    0   N/A  N/A       837      G   /usr/lib/xorg/Xorg                 83MiB |
|    0   N/A  N/A      1259      G   cinnamon                           24MiB |
+-----------------------------------------------------------------------------+

 uname -r
5.4.0-104-generic

Os is Linux Mint 20.3 (based on Ubuntu 20.04)

File: tests/test_endpoint_correction.py [hunk 2]
```diff
@@ -187,7 +187,7 @@ def test_endpoint_correction():
 
     assert results[0] < 0.15
     assert results[1] > 0.30
-    assert results[2] < 0.25
+    assert results[2] < 0.30
```
[the remark this answers is not in the document]
Comment (jkausrelay) [reply tests/test_endpoint_correction.py:190]: Yes, jax 0.2.27 and jaxlib 0.1.75

[the remark this answers is not in the document]
Comment (badisa) [reply tests/test_endpoint_correction.py:190]: Hmm that is surprising, but not sure how important this assertion is.

**Review by badisa — APPROVED** [review]

**Comment by jkausrelay** [comment]
Added a fix for the new desktop with two gpus: 
GPU 0: NVIDIA RTX A4000
GPU 1: Quadro P2000 

The get_gpu_count was not using CUDA_VISIBLE_DEVICES, which resulted in errors like:
`GPUassert: no kernel image is available for execution on the device /data/relay/github/timemachine/timemachine/cpp/src/harmonic_bond.cu 89`

File: tests/test_endpoint_correction.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (proteneer) [reply tests/test_endpoint_correction.py:190]: Note super important but still surprising

**Pull request merged**